src/rebar3.erl

```diff
@@ -272,4 +272,14 @@ start_and_load_apps() ->
     application:start(asn1),
     application:start(public_key),
     application:start(ssl),
-    inets:start().
+    inets:start(),
+    inets:start(httpc, [{profile, hex}]),
```

Comment:
That is exactly why it should be named `hex` :). The hex plugin also sets http options (like proxy settings) based on config values in `~/.hex/hex.config` which we wouldn't necessarily want to use for other stuff in rebar3.